lib/main.js

```diff
@@ -141,49 +138,61 @@ HTMLUglify.prototype.processRules = function(rules, lookups) {
   });
 };
 
-HTMLUglify.prototype.rewriteElements = function($, lookups) {
+HTMLUglify.prototype.rewriteElements = function(tree, lookups) {
   var self = this;
 
   lookups = lookups || {};
 
-  $('*[id]').each(function() {
-    self.pointerizeIdAndFor('id', $(this), lookups);
-  });
+  return tree.walk(function(node) {
+    if (node.attrs) {
+      if (node.attrs.class) {
+        self.pointerizeClass(node, lookups);
+      }
 
-  $('*[for]').each(function() {
-    self.pointerizeIdAndFor('for', $(this), lookups);
-  });
+      if (node.attrs.id) {
+        self.pointerizeIdAndFor('id', node, lookups);
+      }
 
-  $('*[class]').each(function() {
-    self.pointerizeClass($(this), lookups);
+      if (node.attrs.for) {
+        self.pointerizeIdAndFor('for', node, lookups);
+      }
+    }
+    return node;
   });
-
-  return $;
 };
 
-HTMLUglify.prototype.rewriteStyles = function($, lookups) {
+HTMLUglify.prototype.rewriteStyles = function(tree, lookups) {
   var self = this;
 
   lookups = lookups || {};
 
-  $('style').each(function() {
-    var $style = $(this);
-    var ast = postcssSafeParser($style.text());
-    self.processRules(ast.nodes, lookups);
-    $style.text(ast.toString());
+  return tree.walk(function(node) {
```

Comment:
in this case, about the same. 2 if vs  https://github.com/posthtml/posthtml/blob/master/lib/api.js#L73-L94